docs: add tip on browser extensions to perf guide (#15577)
vitejs/vite: docs/guide/performance.md

- Slow page loads
- Slow builds

## Avoid Browser Extensions

> Some browser extensions may interfere with requests and slow down startup and reload times for large apps, especially when using browser dev tools. We recommend creating a dev-only profile without extensions, or switch to incognito mode, while using Vite's dev server in these cases.

Reviewer: ```suggestion
Some browser extensions may interfere with requests and slow down startup and reload times for large apps, especially when using browser dev tools. We recommend creating a dev-only profile without extensions, or switch to incognito mode, while using Vite's dev server in these cases. Incognito mode should also be faster than a regular profile without extensions.
```

> Some browser extensions may interfere with requests and slow down startup and reload times for large apps, especially when using browser dev tools. We recommend creating a dev-only profile without extensions while using Vite's dev server in these cases. Another option is using Incognito mode, that should also provide an extra speed up when loading your app.

Reviewer 2: ```suggestion
Some browser extensions may interfere with requests and slow down startup and reload times for large apps, especially when using browser dev tools. We recommend creating a dev-only profile without extensions, or switch to incognito mode, while using Vite's dev server in these cases.
```

Maybe this is a bit more concise. Technically, extensions can also be enabled in incognito mode, but I'm not really sure how to elaborate that. Maybe assuming incognito mode has no extensions is fine.

Reviewer: I would also like to transmit that Incognito mode is even faster that a normal profile without extensions. That is why I had it separately. Maybe like this? Or is it too long?
> Some browser extensions may interfere with requests and slow down startup and reload times for large apps, especially when using browser dev tools. We recommend creating a dev-only profile without extensions, or switch to incognito mode, while using Vite's dev server in these cases. Incognito mode should also be faster than a regular profile, even without extensions.

Reviewer 2: That makes sense to me. I'd probably tweak the last sentence slightly as:

> Incognito mode should also be faster than a regular profile without extensions.
